File: src/main/java/tk/hiddenname/smarthome/entity/task/processing/objects/SetDigitalSignalObject.java
package tk.hiddenname.smarthome.entity.task.processing.objects;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import tk.hiddenname.smarthome.service.task.impl.processor.Processor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Data
@Entity
@NoArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Table(name = "processing_set_digital_signal")
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public abstract class SetDigitalSignalObject extends ProcessingObject implements Processor {

    @NotNull
    @Column(name = "device_id", nullable = false)
    private Integer deviceId;

    @Column(nullable = false)
    private int delay = 0;

    @NotNull
    @Column(nullable = false)
    private boolean targetState;
}

File: src/main/kotlin/tk/hiddenname/smarthome/model/task/processing/objects/SetDigitalSignalObject.kt
package tk.hiddenname.smarthome.model.task.processing.objects

import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import tk.hiddenname.smarthome.model.task.TaskValidationGroup
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Table
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@Entity
@Table(name = "processing_set_digital_signal")
@JsonNaming(SnakeCaseStrategy::class)
class SetDigitalSignalObject(
        @field:Min(1, groups = [TaskValidationGroup::class])
        @field:NotNull(message = "device id should be specified", groups = [TaskValidationGroup::class])
        @Column(name = "device_id", nullable = false, updatable = false)
        val deviceId: Long? = null,

        @field:Min(0, groups = [TaskValidationGroup::class])
        @field:Max(86400, groups = [TaskValidationGroup::class])
        @Column(nullable = false)
        val delay: Int = 0,

        @field:NotNull(message = "target digital state should be specified", groups = [TaskValidationGroup::class])
        @Column(name = "target_state", nullable = false)
        val targetState: Boolean? = null
) : ProcessingObject() {
    override fun toString(): String {
        return "SetDigitalSignalObject(deviceId=$deviceId, delay=$delay, targetState=$targetState)"
    }
}